### PR TITLE
expand paths and split PATH in `std path add`

### PR DESCRIPTION
**Pull request by amtoine**
related to
- https://discord.com/channels/601130461678272522/614593951969574961/1162406310155923626

# Description
this PR
- does a bit of minor refactoring
- makes sure the input paths get expanded
- makes sure the input PATH gets split on ":"
- adds a test
- fixes the other tests

# User-Facing Changes
should give a better overall experience with `std path add`

# Tests + Formatting
adds a new test case to the `path_add` test and fixes the others.

# After Submitting

### CODE REVIEW

File: crates/nu-std/tests/test_std.nu [hunk 1]
```diff
@@ -8,23 +8,24 @@ def path_add [] {
 
     with-env [$path_name []] {
         def get_path [] { $env | get $path_name }
+        def expand_paths []: list<string> -> list<path> { each { path expand } }
```
Comment (weirdan) [inline crates/nu-std/tests/test_std.nu:11]: You don't _need_ `each` here; `path expand` works with lists just fine. Considering this, you don't need this function at all, as you can replace calls to it with just `path expand`.

Comment (amtoine) [reply crates/nu-std/tests/test_std.nu:11]: ooh wow, this is very true, good catch :pray:

Comment (amtoine) [reply crates/nu-std/tests/test_std.nu]: should be good to go in 2eaad1ff6 :wink:

**Comment by amtoine** [comment]
yeah, the Windows CI is finally happy :partying_face:

**Review by weirdan — APPROVED** [review]

**Comment by amtoine** [comment]
this should address the original issue in Discord.
thanks @weirdan for the review, let's try that :relieved:

**Pull request merged**